### PR TITLE
feat: add `pixi global install`

### PR DESCRIPTION
**Pull request by Hofer-Julian**

### CODE REVIEW

**Comment by ruben-arts** [comment]
The latest changes are done to refactor the project some more to also do the binary cleaning per environment instead of in general. This should also fix the issue we had with the integration tests where we wouldn't delete old binaries in `pixi g install`

**Comment by ruben-arts** [comment]
I'll fix the tests today.

**Comment by ruben-arts** [comment]
The main differences between @Hofer-Julian's first implementation and the refactored version is that we now don't sync on install we only install the requested environments and this also cleanup those.

File: src/global/project/environment.rs [hunk 1]
```diff
@@ -64,3 +63,37 @@ pub struct ParseEnvironmentNameError {
...
+//         &self.name
+//     }
+//
+//     /// Returns the prefix of the environment.
+//     pub fn prefix(&self) -> &Prefix {
+//         &self.prefix
+//     }
+//
+//
+// }
```
Comment (baszalmstra) [inline src/global/project/environment.rs:99]: Can this be removed?

**Review by ruben-arts — APPROVED** [review]
Let's go :rocket:

**Pull request merged**